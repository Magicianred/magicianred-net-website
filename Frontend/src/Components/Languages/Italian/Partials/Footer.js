import React from 'react';

const appName = process.env.REACT_APP_NAME;
const appVersion = process.env.REACT_APP_VERSION;

/**
 * Functional Component for showing the footer of the site
 * @component
 * @example
 * return (
 *  <Footer />
 * )
 */
const Footer = () => {
    return (
        <footer>
		    <section id="footer-area">
    			<section id="footer-outer-block">
					<aside class="footer-segment">
					    <h4>Friends</h4>
                        <ul>
                            <li><a href="#">one linkylink</a></li>
                            <li><a href="#">two linkylinks</a></li>
                            <li><a href="#">three linkylinks</a></li>
                        </ul>
					</aside>
					<aside class="footer-segment">
                        <h4>Awesome Stuff</h4>
                        <ul>
                            <li><a href="#">one linkylink</a></li>
                            <li><a href="#">two linkylinks</a></li>
                            <li><a href="#">three linkylinks</a></li>
                        </ul>
					</aside>
					<aside class="footer-segment">
                        <h4>Coolness</h4>
                        <ul>
                            <li><a href="#">one linkylink</a></li>
                            <li><a href="#">two linkylinks</a></li>
                            <li><a href="#">three linkylinks</a></li>
                        </ul>
					</aside>
					<aside class="footer-segment">
                        <h4>Licenza Template</h4>
                        <p><small>This work by <i>freehtml5templates.com</i> is licensed under a
                        <i>Creative Commons Attribution 3.0</i> United States License.</small></p>
					</aside>
			    </section>
                {/* <p>
                    <small>This is the footer <i>{appName}</i> version <b>{appVersion}</b></small>
                </p> */}
            </section>
        </footer>
    )
}

export default Footer;