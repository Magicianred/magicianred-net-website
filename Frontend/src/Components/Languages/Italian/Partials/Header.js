import React from 'react';
import PropTypes from 'prop-types';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import logo from './../../../../logo.svg';

import LanguagesSwitch from '../../../Partials/LanguagesSwitch';

const publicUrl = process.env.PUBLIC_URL;

/**
 * Functional Component for showing the header of the site. 
 * @component
 * @param {string} title Title of the site
 * @example
 * const sitename = 'my site name'
 * return (
 *   <Header title={sitename} />
 * )
 */
const Header = ({ title }) => {
    return (
        <header className="partial header" style={{ marginBottom: "15px" }}>
            <Navbar className="menu">
                <Navbar.Brand href={`${publicUrl}/#/`}>
                    {/* <img
                        src={logo}
                        width="30"
                        height="30"
                        className="d-inline-block align-top"
                        alt={`${title} logo`}
                    /> */}
                    </Navbar.Brand>
                    <Navbar.Toggle aria-controls="basic-navbar-nav" />
                    <Navbar.Collapse id="basic-navbar-nav">
                        <Nav className="mr-auto">
                            <Nav.Link className="header-link" id="home_link" href={`${publicUrl}/#/it/`}>Home</Nav.Link>
                            <Nav.Link className="header-link" id="web-project_link" href={`${publicUrl}/#/it/progetti-web`}>Progetti Web</Nav.Link>
                            <Nav.Link className="header-link" id="writer-project_link" href={`${publicUrl}/#/it/progetti-scrittura`}>Progetti Scrittura</Nav.Link>
                            <Nav.Link className="header-link" id="permacultura_link" href={`${publicUrl}/#/it/permacultura`}>Permacultura</Nav.Link>
                            <Nav.Link className="header-link" id="about_link" href={`${publicUrl}/#/it/about`}>About</Nav.Link>
                        </Nav>
                        <LanguagesSwitch />
                    </Navbar.Collapse>
            </Navbar>
            <section className="header-description">
                <div id="plandesign"><img src="/images/plans.png" alt="" /></div>
                <h1><a href="#">La Magia delle Idee <small><em>by Magicianred</em></small></a></h1>
                <p>spazio personale per dare libero sfogo alle idee. Non ci saranno solo idee concrete e pienamente sviluppate,
                ma germogli, semi, bozze e quant'altro mi verrà in mente. <br/>
                Molte saranno abbandonate, alcune vedranno la luce e si svilupperanno,
                molte altre invece saranno lasciate libere di circolare con licenze adatte per farle raccoglie a chi se la sentirà.</p>
            </section>
        </header>
    );
}

Header.propTypes = {
    /**
     * Title of the site
     */
    title: PropTypes.string.isRequired,
}

Header.defaultProps = {
    title: 'Default Site title'
}

export default Header;