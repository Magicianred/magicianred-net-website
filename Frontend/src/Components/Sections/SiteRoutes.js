import React from 'react';
import { Route, Switch } from 'react-router-dom';
import ItalianSection from './ItalianSection';
// import EnglishSection from './EnglishSection';
// import SpanishSection from './SpanishSection';
// import GermanSection from './GermanSection';
import Error from '../Languages/Italian/Pages/Error';
import NotFound from '../Languages/Italian/Pages/Error';

/**
 * Functional Component for handle routes of sites.
 * 
 * @component
 * @example
 * return (
 *   <SiteRoutes />
 * )
 */
const SiteRoutes = () =>
    <Switch>
        {/* <Route path={`/en`} component={EnglishSection} />
        <Route path={`/es`} component={SpanishSection} />
        <Route path={`/de`} component={GermanSection} /> */}
        <Route path={`/`} component={ItalianSection} />
        { /* default language */ }
        <Route exact path="/error" component={Error} />
        
        <Route path="*" component={NotFound} />
    </Switch>

export default SiteRoutes;