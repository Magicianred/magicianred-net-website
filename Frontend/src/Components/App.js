import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import Container from 'react-bootstrap/Container';
import SiteRoutes from './Sections/SiteRoutes';
import './App.css';

/**
 * Title of the site
 */
const siteName = 'Simple App React';
const publicUrl = process.env.PUBLIC_URL;

/**
 * This fuction startup the application
 * @function App
 */
function App() {
  return (
    <div id="wrapper">
        <Router basename={publicUrl}>
            <SiteRoutes />
        </Router>
    </div>
  );
}

export default App;
