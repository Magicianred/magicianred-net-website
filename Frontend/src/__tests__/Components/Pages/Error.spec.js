import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import { create } from 'react-test-renderer';
import { MemoryRouter } from "react-router-dom";

import SiteRoutes from "./../../../Components/Sections/SiteRoutes";
import NotFound from "./../../../Components/Pages/NotFound";
import App from "./../../../Components/App";
import Header from "./../../../Components/Partials/Header";
import Footer from "./../../../Components/Partials/Footer";

Enzyme.configure({ adapter: new Adapter() });
let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe("Show the Error page", () => {

    describe("Snapshots", () => {
    
        xit("Matches the snapshot (with default title)", () => {
            const instance = create(<NotFound />);
            expect(instance.toJSON()).toMatchSnapshot();
        });
    
    });

    xit("Show Error page", () => {
        act(() => {
            render(<NotFound />, container);
        });
        expect(container.querySelector("h2").textContent).toBe("404 Page not found");
        expect(container.querySelector("p").textContent).toBe("The page is not found");
    });
    
    xit("Show the header", () => {
        const wrapper = shallow(
            <App />
        );
        expect(wrapper.contains(<Header />)).toBe(true);
    });

    xit("Show the site menu", () => {
    });

    xit("Show the site footer", () => {
    });

});