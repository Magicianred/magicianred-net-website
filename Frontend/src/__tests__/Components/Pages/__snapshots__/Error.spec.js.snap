// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show the Error page Snapshots Matches the snapshot (with default title) 1`] = `
<div
  className="page"
>
  <div
    className="page notfound jumbotron"
  >
    <h2>
      404 Page not found
    </h2>
    <p>
      The page is not found
    </p>
  </div>
</div>
`;
